### DISCUSSION TITLE
CSS files failed to be loaded;net::ERR_TOO_MANY_REDIRECTS

### DISCUSSION THREAD

**Discussion by 0MAL** [mosparo/mosparo#202] [Problems & Issues] [answered]
<img width="1427" alt="toomanyredirect" src="https://github.com/mosparo/mosparo/assets/91258926/151455e8-1fd8-43f7-b407-5796cd1e7fa4">
<img width="1430" alt="redirectnetwork" src="https://github.com/mosparo/mosparo/assets/91258926/ecf33e35-a76e-41c9-8b0a-4b8beacc62ba">

StyleSheet loading fails for the both case below:
`<link href="http://hello-lando.lndo.site/resources/UIID.css" rel="stylesheet">
`
OR/AND
`<script src="https://mosparo.example.com/build/mosparo-frontend.js" defer></script>
<script>
    var m;
    window.onload = function(){
        m = new mosparo('mosparo-box', 'https://mosparo.example.com', '<uuid>', '<publicKey>', {
            loadCssResource: true
        });
    };
</script>`

Only the following code works, but it's mono-color:
`<link href="http://hello-lando.lndo.site/build/mosparo-frontend.css" rel="stylesheet">`

<img width="1427" alt="defaultcss" src="https://github.com/mosparo/mosparo/assets/91258926/989e3ec3-e27d-49f1-898c-a3b2ca270a3b">
<img width="1436" alt="defaultcssnetwork" src="https://github.com/mosparo/mosparo/assets/91258926/85980d0a-288a-41ca-824f-b1489946417b">

**Comment by zepich** [comment]
Hi @0MAL 

Thank you very much for your message.

Do you use a Windows server with IIS?

It sounds like the issue that @nadorator had some weeks ago: https://github.com/orgs/mosparo/discussions/195#discussioncomment-8657198

If that is the problem, I will set up a Windows server and test it to find the issue.

Thank you very much for your response.

Kind regards,

zepich

**Comment by 0MAL** [comment]
No. This error occurs at Lando's LAMP dev server. And I haven't tested at production site of apache yet.

**Reply by zepich** [reply]
Hi @0MAL 

Thank you for your response.

Can you check if the CSS file exists in the directory?

There should be a directory `/<path-to-mosparo>/public/resources/<uuid>/` with at least one file named `<designhash>.css`. Do you have the directory and the file in it?

Additionally, there should be a file `/<path-to-mosparo>/public/resources/mappings.php`. Does this file exist as well?

Thank you very much for your feedback.

Kind regards,

zepich

**Reply by 0MAL** [reply, marked as answer]
Both files exist there.
I think I have solved the issue. I changed the .htaccess file just under the parent folder of unzipped mosparo-stable-v1.1.3.zip like this:
before 
```
<IfModule mod_rewrite.c>
    RewriteEngine On

    RewriteBase /

    RewriteRule ^build(.*)$ public/build/$1 [L]
    RewriteRule ^bundles(.*)$ public/bundles/$1 [L]

    RewriteRule ^robots.txt$ public/robots.txt [L]
```

after
```
<IfModule mod_rewrite.c>
    RewriteEngine On

    RewriteBase /

    RewriteRule ^build(.*)$ public/build/$1 [L]
    RewriteRule ^bundles(.*)$ public/bundles/$1 [L]
    RewriteRule ^resources(.*)$ public/resources/$1 [L]

    RewriteRule ^robots.txt$ public/robots.txt [L]
```


<img width="1435" alt="solvedafterhtaccess" src="https://github.com/mosparo/mosparo/assets/91258926/6bf7eb4e-2b02-4a89-83e9-b8c6de3b96fd">

**Comment by zepich** [comment]
Hi @0MAL 

Thank you very much for your report!

I could reproduce the issue, and your solution fixed it perfectly.

I've added your fix to the code. The fix will be deployed with the next version.

Thank you very much for helping to make mosparo better!

Kind regards,

zepich
